src/astUtils/reflection.ts

```diff
@@ -309,32 +309,39 @@ export function isPrintSeparatorExpression(element: any): element is PrintSepara
 }
 
 // BscType reflection
-export function isStringType(value: any): value is StringType {
-    return value?.kind === BscTypeKind.StringType;
+export function isStringType(value: any, includeBoxed = true): value is StringType {
```

Comment:
Shouldn't these be `false` by default, except in the locations where you know it's allowed? Also, i don't see anywhere that you're passing `false`, so the entire codebase will now be supporting boxed by default. Is all of that okay? Feels a little aggressive, but I don't have all the context.

Reply:
yeah.. I was thinking this too (`includeBoxed = false` by default)... But everywhere in the code where I check a primitive type, I was going to have to  add `true` anyway

The places I check for these primitive types are in typeCompatibility checks,  operator checks, etc... and as far as I know, BrightScript basically treats these the same anyway.

Can you think of a time a Boxed type is not allowed, when a non-boxed type would be?

Reply:
You're right. it's probably fine. I'm imagining some future box/not-boxed validations, but you still have the boolean, and in 99.9% of the time, nobody will care. 

So this is fine as-is

Reply:
Actually, I still have a few thoughts on this.

I think what I'd really like to see is something like:
```
function isStringType()
function isRoStringType()
function isStringTypeLike()
```

I know that's a lot more work to add these to all the boxables, but in the future I envision adding some extra opt-in diagnostics to flag boxed/non-boxed interactions in some cases, and overloading this `isStringType` feels just a little "incorrect" since we _are_ actually comparing two separate types. 

Thoughts?

Reply:
awesome.. this is better :) 

I'll make the change